core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala

```diff
@@ -114,7 +114,13 @@ private[deploy] class HadoopFSDelegationTokenProvider(fileSystems: () => Set[Fil
     // We cannot use the tokens generated with renewer yarn. Trying to renew
     // those will fail with an access control issue. So create new tokens with the logged in
     // user as renewer.
-    sparkConf.get(PRINCIPAL).flatMap { renewer =>
+    val user = sparkConf.get(KERBEROS_RENEWAL_CREDENTIALS) match {
+      case "keytab" => sparkConf.get(PRINCIPAL)
```

Comment:
Can't one get the user all the time with `UserGroupInformation.getCurrentUser().getUserName()`?

Reply:
I guess.